File: src/DotNetPad/DotNetPad.Applications/Samples/AutoPropertyInitializers.vb
Imports System

Namespace Waf.DotNetPad.Samples
    Module AutoPropertyInitializers
        Sub Main()
            Dim customer = New Customer(3)
            Console.WriteLine("Customer {0}: {1}, {2}", customer.Id, customer.First, customer.Last)
        End Sub
    End Module

    Class Customer
        Sub New(id As Integer)
            Me.Id = id
        End Sub

        Public ReadOnly Property Id As Integer

        Public Property First As String = "Luke"

        Public Property Last As String = "Skywalker"
    End Class
End Namespace

File: src/DotNetPad/DotNetPad.Applications/Samples/AutoPropertyInitializers.cs
namespace Waf.DotNetPad.Samples;

internal static class AutoPropertyInitializers
{
    private static void Main()
    {
        var customer = new Customer(3);

        Console.WriteLine("Customer {0}: {1}, {2}", customer.Id, customer.First, customer.Last);
    }
}

public class Customer
{
    public Customer(int id)
    {
        Id = id;
    }

    public int Id { get; }

    public string First { get; set; } = "Luke";

    public string Last { get; set; } = "Skywalker";
}
